monai/apps/reconstruction/networks/nets/complex_unet.py

```diff
@@ -0,0 +1,172 @@
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+from typing import Sequence, Union
+
+import torch.nn as nn
+from torch import Tensor
+from torch.nn import functional as F
+
+from monai.networks.nets.basic_unet import BasicUNet
+
+
+class ComplexUnet(nn.Module):
+    """
+    This variant of U-Net handles complex-value input/output. It can be
+    used as a model to learn sensitivity maps in multi-coil MRI data. It is
+    built based on :py:class:`monai.networks.nets.BasicUNet`.
+    It also applies default normalization to the input which makes it more stable to train.
+
+    The data being a (complex) 2-channel tensor is a requirement for using this model.
+
+    Modified and adopted from: https://github.com/facebookresearch/fastMRI
+
+    Args:
+        spatial_dims: number of spatial dimensions.
+        features: six integers as numbers of features. denotes number of channels in each layer.
+        act: activation type and arguments. Defaults to LeakyReLU.
+        norm: feature normalization type and arguments. Defaults to instance norm.
+        bias: whether to have a bias term in convolution blocks. Defaults to True.
+        dropout: dropout ratio. Defaults to 0.0.
+        upsample: upsampling mode, available options are
+            ``"deconv"``, ``"pixelshuffle"``, ``"nontrainable"``.
+    """
+
+    def __init__(
+        self,
+        spatial_dims: int = 2,
+        features: Sequence[int] = (32, 32, 64, 128, 256, 32),
+        act: Union[str, tuple] = ("LeakyReLU", {"negative_slope": 0.1, "inplace": True}),
+        norm: Union[str, tuple] = ("instance", {"affine": True}),
+        bias: bool = True,
+        dropout: Union[float, tuple] = 0.0,
+        upsample: str = "deconv",
+    ):
+        super().__init__()
+        self.unet = BasicUNet(
+            spatial_dims=spatial_dims,
+            in_channels=2,
+            out_channels=2,
+            features=features,
+            act=act,
+            norm=norm,
+            bias=bias,
+            dropout=dropout,
+            upsample=upsample,
+        )
+
+    def complex_to_chan_dim(self, x: Tensor) -> Tensor:
+        """
+        Swaps the complex dimension with the channel dimension so that the network treats real/imaginary
+        parts as two separate channels.
+
+        Args:
+            x: input of shape (B,C,H,W,2)
+
+        Returns:
+            output of shape (B,C*2,H,W)
+        """
+        b, c, h, w, two = x.shape
+        if x.shape[-1] != 2:
+            raise ValueError(f"last dim must be 2, but x.shape[-1] is {x.shape[-1]}.")
+        return x.permute(0, 4, 1, 2, 3).contiguous().view(b, 2 * c, h, w)
+
+    def chan_complex_to_last_dim(self, x: Tensor) -> Tensor:
```

Comment:
Do you feel there will be other recon networks using this func? I was not sure. If you feel so, maybe add these funcs to `complex_utils.py`